gcp/table_gcp_kms_key_version.go

```diff
@@ -0,0 +1,315 @@
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("DestroyEventTime").NullIfZero(),
+			},
+			{
+				Name:        "destroy_time",
+				Description: "The time this CryptoKeyVersion's key material is scheduled for destruction.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("DestroyTime").NullIfZero(),
+			},
+			{
+				Name:        "external_key_uri",
+				Description: "The URI for an external resource that this CryptoKeyVersion represents.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ExternalProtectionLevelOptions.ExternalKeyUri"),
+			},
+			{
+				Name:        "generate_time",
+				Description: "The time this CryptoKeyVersion's key material was generated.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "import_failure_reason",
+				Description: "The root cause of an import failure.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "import_job",
+				Description: "The name of the ImportJob used to import this CryptoKeyVersion.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "protection_level",
+				Description: "The ProtectionLevel describing how crypto operations are performed with this CryptoKeyVersion.",
+				Type:        proto.ColumnType_STRING,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromP(kmsKeyVersionTurbotData, "ResourceName"),
+			},
+			{
+				Name:        "akas",
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromP(kmsKeyVersionTurbotData, "Akas"),
+			},
+
+			// GCP standard columns
+			{
+				Name:        "location",
+				Description: ColumnDescriptionLocation,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromP(kmsKeyVersionTurbotData, "Location"),
+			},
+			{
+				Name:        "project",
+				Description: ColumnDescriptionProject,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromP(kmsKeyVersionTurbotData, "Project"),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTIONS
+
+func listKeyVersionDetails(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("listKeyVersionDetails")
+
+	// Create Service Connection
+	service, err := KMSService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	// Fetch list of Key Details from KeyRings
+	cryptoKeys, err := listKeyDetailsForRings(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+
+	pageSize := types.Int64(1000)
+	limit := d.QueryContext.Limit
+	if d.QueryContext.Limit != nil {
+		if *limit < *pageSize {
+			pageSize = limit
+		}
+	}
+
+	// Split CryptoKeys in chunks of 50 for concurrent fetching of CryptoKeyVersions
+	chunksCryptoKeys := chunkSlice(cryptoKeys, 50)
```

Comment:
As discussed please update the list call to use async call in CryptoKeys.List API itself.